src/preamble.js

```diff
@@ -432,70 +432,56 @@ function stringToAscii(str, outPtr) {
 var UTF8Decoder = typeof TextDecoder !== 'undefined' ? new TextDecoder('utf8') : undefined;
 #endif
 function UTF8ArrayToString(u8Array, idx, sizeInBytes) {
-
-  var endIdx = idx;
-
-  if (typeof sizeInBytes === 'undefined')
-    while (u8Array[endIdx]) ++endIdx;
-  else
-    endIdx += sizeInBytes; //Length info provided externally, e. g. when embedded null bytes should be processed
+    //based on https://gist.github.com/pascaldekloe/62546103a1576803dade9269ccf76330
```

Comment:
I don't see an explicit license on that code. Might need to get permission from @pascaldekloe or write the algorithm in your own words?

Reply:
The code is public domain. Feel free to use it anyway you like. Thanks for the credits though. :-)